cime_config/machines/cmake_macros/amdclanggpu_crusher.cmake

```diff
@@ -0,0 +1,21 @@
+if (NOT DEBUG)
+  string(APPEND CFLAGS   " -O2")
+  string(APPEND CXXFLAGS " -O2")
+  string(APPEND FFLAGS   " -O2")
+endif()
+
+set(PIO_FILESYSTEM_HINTS "gpfs")
+set(NETCDF_PATH "$ENV{NETCDF_DIR}")
+set(PNETCDF_PATH "$ENV{PNETCDF_DIR}")
+string(APPEND CMAKE_OPTS " -DPIO_ENABLE_TOOLS:BOOL=OFF")
+string(APPEND SLIBS " -L$ENV{CRAY_LIBSCI_PREFIX_DIR}/lib -lsci_amd")
+
+set(MPICXX "hipcc")
+set(SCXX "hipcc")
+string(APPEND CXXFLAGS " -I$ENV{MPICH_DIR}/include --amdgpu-target=gfx90a")
+string(APPEND SLIBS    " -L$ENV{MPICH_DIR}/lib -lmpi -L$ENV{CRAY_MPICH_ROOTDIR}/gtl/lib -lmpi_gtl_hsa")
+
+string(APPEND KOKKOS_OPTIONS " -DKokkos_ENABLE_HIP=On -DKokkos_ARCH_ZEN3=On -DKokkos_ARCH_VEGA90A=On")
+
+set(USE_HIP "TRUE")
+string(APPEND HIP_FLAGS "-munsafe-fp-atomics -D__HIP_ROCclr__ -D__HIP_ARCH_GFX90A__=1 --rocm-path=$ENV{ROCM_PATH} --offload-arch=gfx90a -x hip")
```

Comment:
Yes, I will check. Btw, I think some of the missing library errors are probably due to using hipcc instead of CC. 

Remind me, why we don't use CC with necessary HIP flags and doing it the other way around?

Reply:
Looks like Xingqiu is using CC etc. in his KPP branch for MMF runs
https://github.com/E3SM-Project/E3SM/blob/whannah/mmf/KPP-dev/cime_config/machines/cmake_macros/amdclanggpu_frontier.cmake#L12

Reply:
@amametjanov Have you run the MMF KPP cases on Crusher using crayclangpu or just on the CPU?

Reply:
This was raised in Crusher GPU PR review: that SCREAM is using hipcc, not CC. Maybe @oksanaguba can confirm?

E3SM nightly test `SMS_Ln9.ne4pg2_ne4pg2.F2010-MMF1.crusher_crayclanggpu` with `hipcc` is now reporting to https://my.cdash.org/index.php?project=E3SM under `e3sm_gpucxx_next_crayclanggpu `

Reply:
I found an earlier thread in Slack (frontier channel) where @twhite-cray recommended sticking to hipcc for CXX targets. So, hipcc makes sense. Later, we need to get Xingqui's KPP problem working with this setup.